src/utils/client.go

```diff
@@ -0,0 +1,264 @@
+	// Configure the oidc setting in Harbor with current uaa.json
+	// utils -config-oidc -harbor-server https://10.186.197.207 -password Harbor12345 -uaa-server https://api.pks.local  -verify-cert
+
+	isShowUAA := flag.Bool("show-uaa", false, "Display UAA json which add token salt info")
+	trustedCA := flag.String("trusted-ca", "/var/vcap/jobs/harbor/config/ca.crt", "Trusted certificate")
+	insecure := flag.Bool("insecure", false, "insecure")
+	isConfigUAA := flag.Bool("config-uaa", false, "Config UAA")
+	isConfigOIDC := flag.Bool("config-oidc", false, "Config OIDC")
+	harborServerUrl := flag.String("harbor-server", "", "Harbor server url to configure")
+	uaaJsonPath := flag.String("uaa-json", "/var/vcap/jobs/harbor/config/uaa.json", "uaa json file path")
+	password := flag.String("password", "", "password to configure harbor")
+	tokenSaltFile := flag.String("salt-file", "/data/secret/keys/uaa_token_salt", "uaa token salt file")
+	uaaServer := flag.String("uaa-server", "", "UAA server to authenticate")
+	verifyCert := flag.Bool("verify-cert", false, "Verify cert")
+	// verbose is used for debugging, dump all information to console
+	verbose := flag.Bool("verbose", false, "Display verbose info")
+
+	flag.Parse()
+
+	trace = *verbose
+
+	if *isShowUAA == true {
+		uaaProf, err := parseJsonFile(*uaaJsonPath)
+		checkError(err)
+		uaaProf.TokenSalt = readTokenSalt(*tokenSaltFile)
+		Trace("token salt is changed to " + uaaProf.TokenSalt)
+		content, err := json.Marshal(uaaProf)
+		fmt.Print(string(content))
+		os.Exit(0)
+	}
+
+	checkParameter(harborServerUrl, password, uaaServer)
+	client := createHttpClient(*trustedCA, *insecure)
+	c := newConfigClient(client, *harborServerUrl, "admin", *password)
+	uaaProf, err := parseJsonFile(*uaaJsonPath)
+	checkError(err)
+	if *isConfigOIDC == true {
+		err = c.configOIDC(uaaProf, *uaaServer, *verifyCert)
+	}
+
+	if *isConfigUAA == true {
+		err = c.configUAA(uaaProf, *uaaServer, *verifyCert)
+
+	}
+	checkError(err)
+	os.Exit(0)
+}
+
+func checkParameter(harborServerUrl, password, uaaServer *string) {
+	if len(*harborServerUrl) == 0 {
+		Trace("Please provide server url to configure")
+		os.Exit(1)
+	}
+	if len(*password) == 0 {
+		Trace("need to provide admin password to configure")
+		os.Exit(1)
+	}
+	if len(*uaaServer) == 0 {
+		Trace("Need to provide the uaa server info")
+		os.Exit(1)
+	}
+}
+
+func checkError(err error) {
+	if err != nil {
+		Trace(err)
+		os.Exit(1)
+	}
+}
+
+func parseJsonFile(jsonPath string) (*UAAProfile, error) {
+	var profile UAAProfile
+	content, err := ioutil.ReadFile(jsonPath)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(content, &profile)
+	if err != nil {
+		return nil, err
+	}
+	return &profile, nil
+}
+
+func readTokenSalt(tokenSaltFile string) string {
+	var token string
+	f, err := os.Open(tokenSaltFile)
+	if os.IsNotExist(err) {
+		token, err = tokenSalt(tokenSaltFile)
+		if err != nil {
```

Comment:
How do you know the error if it's eaten?

Reply:
Updated